dags/gmail_data_import_pipeline.py

```diff
@@ -0,0 +1,155 @@
+    GmailGetDataConfig
+)
+
+from data_pipeline.utils.pipeline_file_io import (
+    get_yaml_file_as_dict
+)
+
+from data_pipeline.gmail_data.get_gmail_data import (
+    get_gmail_service_for_user_id,
+    get_label_list,
+    write_dataframe_to_file
+)
+
+GMAIL_SCOPES = ['https://www.googleapis.com/auth/gmail.readonly']
+GMAIL_ACCOUNT_SECRET_FILE_ENV = 'GOOGLE_APPLICATION_CREDENTIALS'
+
+GMAIL_DATA_USER_ID_ENV = "GMAIL_DATA_USER_ID"
+GMAIL_DATA_CONFIG_FILE_PATH_ENV_NAME = "GMAIL_DATA_CONFIG_FILE_PATH"
+DEPLOYMENT_ENV_ENV_NAME = "DEPLOYMENT_ENV"
+DEFAULT_DEPLOYMENT_ENV = "ci"
+
+DAG_ID = "Gmail_Data_Import_Pipeline"
+
+LOGGER = logging.getLogger(__name__)
+
+
+def get_env_var_or_use_default(env_var_name, default_value=None):
+    return os.getenv(env_var_name, default_value)
+
+
+def get_data_config(**kwargs):
+    conf_file_path = get_env_var_or_use_default(
+        GMAIL_DATA_CONFIG_FILE_PATH_ENV_NAME, ""
+    )
+    LOGGER.info('conf_file_path: %s', conf_file_path)
+    data_config_dict = get_yaml_file_as_dict(conf_file_path)
+    LOGGER.info('data_config_dict: %s', data_config_dict)
+    kwargs["ti"].xcom_push(
+        key="data_config_dict",
+        value=data_config_dict
+    )
+
+
+def data_config_from_xcom(context):
+    dag_context = context["ti"]
+    data_config_dict = dag_context.xcom_pull(
+        key="data_config_dict", task_ids="get_data_config"
+    )
+    LOGGER.info('data_config_dict: %s', data_config_dict)
+    deployment_env = get_env_var_or_use_default(
+        DEPLOYMENT_ENV_ENV_NAME, DEFAULT_DEPLOYMENT_ENV)
+    data_config = GmailGetDataConfig(
+        data_config_dict, deployment_env)
+    return data_config
+
+
+def get_gmail_service():
+    secret_file = get_env_var_or_use_default(GMAIL_ACCOUNT_SECRET_FILE_ENV, "")
+    user_id = get_env_var_or_use_default(GMAIL_DATA_USER_ID_ENV, "")
+    return get_gmail_service_for_user_id(
+        secret_file=secret_file,
+        scopes=GMAIL_SCOPES,
+        user_id=user_id
+    )
+
+
+def gmail_label_data_etl(**kwargs):
+    data_config = data_config_from_xcom(kwargs)
+    user_id = get_env_var_or_use_default(GMAIL_DATA_USER_ID_ENV, "")
+
+    with TemporaryDirectory() as tmp_dir:
+        filename = os.path.join(tmp_dir, data_config.stage_file_name_labels)
+
+        write_dataframe_to_file(
+            df_data_to_write=get_label_list(get_gmail_service(),  user_id),
+            target_file_path=filename
+        )
+
+        LOGGER.info('Created file: %s', filename)
+
+        generated_schema = generate_schema_from_file(filename)
+        LOGGER.info('generated_schema: %s', generated_schema)
+
+        create_table_if_not_exist(
+            project_name=data_config.project_name,
+            dataset_name=data_config.dataset,
+            table_name=data_config.table_name_labels,
+            json_schema=generated_schema
+        )
+        LOGGER.info('Created table: %s', data_config.table_name_labels)
+
+        if filename.lower().endswith('.csv'):
```

Comment:
Currently `generate_schema_from_file` (called above) doesn't seem to support CSV files. I would maybe remove the CSV handling as it would probably fail now?